internal/certs/cert_manager.go

```diff
@@ -0,0 +1,177 @@
+package certs
+
+import (
+	"crypto/x509"
+	"io/ioutil"
+	"os"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ory/oathkeeper/driver/configuration"
+	"github.com/pkg/errors"
+)
+
+type CertManager struct {
+	c configuration.Provider
+
+	// isWindows is set to true if the application is running in a Windows
+	// environment. This information is used to decide whether the stystem
+	// certificate pool should be loaded or not as Windows does not support
+	// this feature and will return an error.
+	isWindows bool
+
+	// cachePool is used to optimize performance and to avoid reloading the cert
+	// pool at every request. The cache expires after its time-to-live has ended
+	// or the an external action has invalidated the cache itself.
+	cachePool *x509.CertPool
+
+	// cacheTime contains the last update of the cache and is used to check
+	// whether it has reached the end of its time-to-live.
+	cacheTime time.Time
+
+	// cacheCerts
+	cacheCerts sync.Map
+
+	requests int64
+}
+
+type CertCache struct {
+	path string
+	stat *os.FileInfo
+}
+
+func NewCertManager(c configuration.Provider) *CertManager {
+	return &CertManager{
+		c:         c,
+		isWindows: runtime.GOOS == "windows",
+	}
+}
+
+func (cm *CertManager) systemCertPool() (*x509.CertPool, error) {
+	if cm.isWindows {
+		return x509.NewCertPool(), nil
```

Comment:
Does windows still use the system cert pool though?

Reply:
Windows does not support the system certificate pool. Calling `SystemCertPool` will always return the error `errors.New("crypto/x509: system root pool is not available on Windows")`

https://golang.org/src/crypto/x509/cert_pool.go?s=3079:3119#L95